src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsCard.component.ts

```diff
@@ -0,0 +1,125 @@
+import { DEFAULT_VIEW_CARD_HEIGHT, DEFAULT_VIEW_CARD_WIDTH } from 'sql/workbench/services/notebook/browser/notebookViews/notebookViewModel';
+import { NotebookViewsExtension } from 'sql/workbench/services/notebook/browser/notebookViews/notebookViewsExtension';
+import { CellChangeEventType, INotebookView, INotebookViewCellMetadata } from 'sql/workbench/services/notebook/browser/notebookViews/notebookViews';
+
+@Component({
+	selector: 'view-card-component',
+	templateUrl: decodeURI(require.toUrl('./notebookViewsCard.component.html'))
+})
+export class NotebookViewsCardComponent implements OnInit {
+	private _metadata: INotebookViewCellMetadata;
+	private _activeView: INotebookView;
+
+	@Input() cell: ICellModel;
+	@Input() model: NotebookModel;
+	@Input() views: NotebookViewsExtension;
+	@Input() ready: boolean;
+	@Output() onChange: EventEmitter<any> = new EventEmitter();
+
+	@ViewChild('templateRef') templateRef: TemplateRef<any>;
+	@ViewChild('item', { read: ElementRef }) private _item: ElementRef;
+
+	constructor(
+		@Inject(forwardRef(() => ChangeDetectorRef)) private _changeRef: ChangeDetectorRef,
+	) { }
+
+	ngOnInit() { }
+
+	ngOnChanges() {
+		if (this.views) {
+			this._activeView = this.views.getActiveView();
+			this._metadata = this.views.getCellMetadata(this.cell);
+		}
+	}
+
+	ngAfterContentInit() {
+		if (this.views) {
+			this._activeView = this.views.getActiveView();
+			this._metadata = this.views.getCellMetadata(this.cell);
+		}
+	}
+
+	ngAfterViewInit() {
+		this.detectChanges();
+	}
+
+	get elementRef(): ElementRef {
+		return this._item;
+	}
+
+	changed(event: CellChangeEventType) {
+		this.onChange.emit({ cell: this.cell, event: event });
+	}
+
+	detectChanges() {
+		this._changeRef.detectChanges();
+	}
+
+	public selectCell(cell: ICellModel, event?: Event) {
+		if (event) {
+			event.stopPropagation();
+		}
+		if (!this.model.activeCell || this.model.activeCell.id !== cell.id) {
+			this.model.updateActiveCell(cell);
+			this.changed('active');
+		}
+	}
+
+	public hide(): void {
+		this.changed('hide');
+	}
+
+	public get data(): any {
+		return this._metadata?.views?.find(v => v.guid === this._activeView.guid);
+	}
+
+	public get width(): number {
+		return this.data?.width ? this.data.width : DEFAULT_VIEW_CARD_WIDTH;
+	}
+
+	public get height(): number {
+		return this.data.height ? this.data.height : DEFAULT_VIEW_CARD_HEIGHT;
+	}
+
+	public get x(): number {
+		return this.data?.x;
+	}
+
+	public get y(): number {
+		return this.data?.y;
+	}
+
+	public get display(): boolean {
+		if (!this._metadata || !this._activeView) {
+			return true;
+		}
+
+		return !this.data?.hidden;
+	}
+
+	public get showActionBar(): boolean {
+		return this.cell.active;
+	}
+}
+
+registerThemingParticipant((theme: IColorTheme, collector: ICssStyleCollector) => {
+	collector.addRule(`
+		.notebook-button.toolbarIconStop {
+			background-color: white;
```

Comment:
Colors generally should be theme-aware. Is a white background color going to look correct on light theme?

Reply:
And if this isn't actually doing anything different with the theme there's no reason to have this be registered as a theming participant. Just put this in the component css at that point.

Reply:
That make sense. I have moved this to a css file. I also removed the white background color. It wasn't needed.